source/io/module_eggShell.js

```diff
@@ -239,6 +247,257 @@
+            validateVisitMethod(visitFn, 'visit' + intName);
+            return visitFn.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitFloat = function (typeVisitor, args) {
+            var typeRef = args[0].typeRef;
+            var sizeOfFloat = Module.eggShell.typeTopAQSize(typeRef);
+            var visitFn;
+            switch (sizeOfFloat) {
+            case 4:
+                visitFn = typeVisitor.visitSingle;
+                break;
+            case 8:
+                visitFn = typeVisitor.visitDouble;
+                break;
+            default:
+                throw new Error('Unexpected size for a Float value');
+            }
+
+            validateVisitMethod(visitFn, 'visit' + (sizeOfFloat === 4 ? 'Single' : 'Double'));
+            return visitFn.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitString = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitString, 'visitString');
+            return typeVisitor.visitString.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitComplex = function (typeVisitor, args) {
+            var typeRef = args[0].typeRef,
+                sizeOfComplex = Module.eggShell.typeTopAQSize(typeRef),
+                visitFn;
+            switch (sizeOfComplex) {
+            case 8:
+                visitFn = typeVisitor.visitComplexSingle;
+                break;
+            case 16:
+                visitFn = typeVisitor.visitComplexDouble;
+                break;
+            default:
+                throw new Error('Unexpected size for a Complex value');
+            }
+
+            validateVisitMethod(visitFn, 'visitComplex' + (sizeOfComplex === 8 ? 'Single' : 'Double'));
+            return visitFn.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitAnalogWaveform = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitAnalogWaveform, 'visitAnalogWaveform');
+            return typeVisitor.visitAnalogWaveform.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitTimestamp = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitTimestamp, 'visitTimestamp');
+            return typeVisitor.visitTimestamp.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitPath = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitPath, 'visitPath');
+            return typeVisitor.visitPath.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitArray = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitArray, 'visitArray');
+            return typeVisitor.visitArray.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitCluster = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitCluster, 'visitCluster');
+            return typeVisitor.visitCluster.apply(typeVisitor, args);
+        };
+
+        var typeDispatchFunctions = [
```

Comment:
Is it a good a idea to put these 2 arrays in a Map?

Reply:
Maps are good if things are added and removed dynamically in the application lifecycle. Since this is static maybe an array of objects?

```
var typeHandlers = [{
    typeChecker: Module.eggShell.typeIsBoolean,
    dispatcher: dispatchVisitBoolean
}]
```

Reply:
Cool, yeah I forgot about the use case of a Map. I'll switch to this one instead.